processor/k8sattributesprocessor/e2e_test.go

```diff
@@ -12,60 +12,207 @@
+				"k8s.labels.app":              newExpectedValue(equal, "test-telemetrygen-daemonset"),
+				"k8s.container.name":          newExpectedValue(equal, "telemetrygen"),
+				"k8s.container.restart_count": newExpectedValue(equal, "0"),
+				"container.image.name":        newExpectedValue(equal, "ghcr.io/open-telemetry/opentelemetry-collector-contrib/telemetrygen"),
+				"container.image.tag":         newExpectedValue(equal, "latest"),
+				"container.id":                newExpectedValue(exist, ""),
+			},
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			scanSpanForAttributes(t, tc.service, tc.attrs)
+		})
+	}
+}
+
+func scanSpanForAttributes(t *testing.T, expectedService string, kvs map[string]*expectedValue) {
+	var err error
+
+	f, _ := os.Open("./testdata/trace.json")
 	defer f.Close()
 
 	scanner := bufio.NewScanner(f)
-
 	const maxCapacity int = 8388608
 	buf := make([]byte, maxCapacity)
 	scanner.Buffer(buf, maxCapacity)
 
 	unmarshaler := ptrace.JSONUnmarshaler{}
-	jobKeyFound := false
-	nsKeyFound := false
 	for scanner.Scan() {
-		traces, err := unmarshaler.UnmarshalTraces(scanner.Bytes())
-		assert.NoError(t, err)
+		traces, serr := unmarshaler.UnmarshalTraces(scanner.Bytes())
+		if serr != nil {
+			continue
+		}
 
 		len := traces.ResourceSpans().Len()
 		for i := 0; i < len; i++ {
 			resourceSpans := traces.ResourceSpans().At(i)
 
-			resourceSpans.Resource().Attributes().Range(
-				func(k string, v pcommon.Value) bool {
-					if k == "k8s.job.name" && v.Str() == "test-telemetrygen" {
-						jobKeyFound = true
-					}
-					if k == "k8s.namespace.name" && v.Str() == "default" {
-						nsKeyFound = true
-					}
-					return true
-				},
-			)
+			service, exist := resourceSpans.Resource().Attributes().Get("service.name")
+			assert.Equal(t, true, exist, "span do not has 'service.name' attribute in resource")
+
+			if service.AsString() != expectedService {
+				continue
+			}
+
+			if err = resourceHasAttributes(resourceSpans.Resource(), kvs); err == nil {
+				return
+			}
 		}
 	}
+	assert.NoError(t, err)
+}
+
+func resourceHasAttributes(resource pcommon.Resource, kvs map[string]*expectedValue) error {
+	exists := make(map[string]bool)
+	for k := range kvs {
+		exists[k] = false
+	}
 
-	assert.True(t, jobKeyFound && nsKeyFound, "job and namespace attributes are not found")
+	resource.Attributes().Range(
+		func(k string, v pcommon.Value) bool {
+			if val, ok := kvs[k]; ok {
+				switch val.mode {
+				case equal:
+					if val.value == v.AsString() {
+						exists[k] = true
+					}
+				case regex:
+					matched, _ := regexp.MatchString(val.value, v.AsString())
+					if matched {
+						exists[k] = true
+					}
+				case exist:
+					exists[k] = true
+				}
+
+			}
+			return true
+		},
+	)
 
-	if err := scanner.Err(); err != nil {
-		assert.NoError(t, err)
+	for k, v := range exists {
+		if !v {
+			return fmt.Errorf("%v attribute not found", k)
+		}
 	}
+	return nil
```

Comment:
Would be nice to accumulate all the not found attributes in one error. You can do

```suggestion
	var err error
	for k, v := range exists {
		if !v {
			err = multierr.Append(err, fmt.Errorf("%v attribute not found", k))
		}
	}
	return err
```